aten/src/ATen/templates/StorageDerived.cpp

```diff
@@ -4,25 +4,46 @@
 
 #include "ATen/Half.h"
 #include "ATen/Allocator.h"
+#include <ATen/Context.h>
 
 #include "ATen/Config.h"
 $extra_cuda_headers
 
 namespace at {
 
-${Storage}::${Storage}():
-    Storage(${THStorage}_new(${state})) {}
-
-${Storage}::${Storage}(THStorage* storage):
-    Storage(storage) {}
+${Storage}::${Storage}() {
+  storage_impl_ = new StorageImpl(
+      ScalarType::${ScalarName}, 
+      0,
+#if ${isCUDA}
+      globalContext().getTHCState()->cudaDeviceAllocator,
+#else
+      getTHDefaultAllocator(),
+#endif
+      StorageImpl::RESIZABLE);
+  storage_impl_->clear_flag(StorageImpl::RESIZABLE);
```

Comment:
This looks wrong. If we used the normal allocator to allocate memory for the storage, it should be resizable. Non-resizable storages are when we are given a pointer directly from some external source.